app/src/main/java/org/wikipedia/notifications/NotificationFilterItemView.kt

```diff
@@ -23,39 +28,73 @@ class NotificationFilterItemView constructor(context: Context, attrs: AttributeS
-            foreground =
-                AppCompatResources.getDrawable(context, ResourceUtil.getThemedAttributeId(context, R.attr.selectableItemBackground))
+            foreground = AppCompatResources.getDrawable(context, ResourceUtil.getThemedAttributeId(context, R.attr.selectableItemBackground))
         }
         setOnClickListener {
-            when (binding.notificationFilterTitle.text.toString()) {
-                context.getString(R.string.wikimedia_commons) -> callback?.onCheckedChanged("commons")
-                context.getString(R.string.wikidata) -> callback?.onCheckedChanged("wikidata")
-                else -> callback?.onCheckedChanged(binding.notificationFilterLanguageCode.text!!.toString())
+            NotificationCategory.FILTERS_GROUP.find { context.getString(it.title) == binding.notificationFilterTitle.text.toString() }?.let {
+                callback?.onCheckedChanged(it.id)
+                return@setOnClickListener
+            } ?: run {
+                when (binding.notificationFilterTitle.text.toString()) {
+                    context.getString(R.string.notifications_all_wikis_text) -> callback?.onCheckedChanged(context.getString(R.string.notifications_all_wikis_text))
+                    context.getString(R.string.notifications_all_types_text) -> callback?.onCheckedChanged(context.getString(R.string.notifications_all_types_text))
+                    context.getString(R.string.wikimedia_commons) -> callback?.onCheckedChanged("commons")
+                    context.getString(R.string.wikidata) -> callback?.onCheckedChanged("wikidata")
+                    else -> callback?.onCheckedChanged(filter?.languageCode.toString())
+                }
             }
         }
-        DeviceUtil.setContextClickAsLongClick(this)
     }
 
-    fun setContents(langCode: String?, title: String, selected: Boolean, imageRes: Int?) {
-        binding.notificationFilterTitle.text = title
-        binding.notificationFilterCheck.visibility = if (selected) View.VISIBLE else View.GONE
-        langCode?.let {
-            binding.notificationFilterLanguageCode.text = langCode
+    fun setContents(filter: Filter) {
+        this.filter = filter
+        binding.notificationFilterTitle.text = getTitleFor(filter.languageCode)
+        binding.notificationFilterCheck.visibility = if (filter.isEnabled()) View.VISIBLE else View.GONE
+        getTitleCodeFor(filter.languageCode)?.let {
+            binding.notificationFilterLanguageCode.text = filter.languageCode
             binding.notificationFilterLanguageCode.visibility = View.VISIBLE
-            ViewUtil.formatLangButton(binding.notificationFilterLanguageCode, langCode, SearchFragment.LANG_BUTTON_TEXT_SIZE_SMALLER, SearchFragment.LANG_BUTTON_TEXT_SIZE_LARGER)
+            ViewUtil.formatLangButton(binding.notificationFilterLanguageCode, it, SearchFragment.LANG_BUTTON_TEXT_SIZE_SMALLER, SearchFragment.LANG_BUTTON_TEXT_SIZE_LARGER)
```

Comment:
Looks like the language code is not correct. Should it be `langCode`?

Reply:
I also notice the text color is incorrect in dark mode, would you update that with a proper attritube color?